src/entity/posts/utils.ts

```diff
@@ -340,6 +341,56 @@ export const createSharePost = async (
   }
 };
 
+export const updateSharePost = async (
+  con: DataSource | EntityManager,
+  userId: string,
+  postId: string,
+  sourceId: string,
+  commentary: string | null,
+) => {
+  let strippedCommentary = commentary;
+
+  if (commentary?.length) {
+    strippedCommentary = commentary.trim();
+    await validateCommentary(strippedCommentary);
+  } else {
+    strippedCommentary = null;
+  }
+
+  try {
+    const mentions = await getMentions(con, commentary, userId, sourceId);
+    const titleHtml = commentary?.length
+      ? generateTitleHtml(commentary, mentions)
+      : null;
+
+    await con.getRepository(SharePost).update(
+      {
+        id: postId,
+        authorId: userId,
```

Comment:
Do we need `authorId` here, `psotId` should be unique?

Reply:
agreed, it's just another layer of security, to avoid the possibility of loading someone else's post, we also scope it by the author and current user ID, we can remove it if you feel it's overkill, since we also have a validation for this

Reply:
From perspective of a function it is called `updateSharePost` so if it was reused to update shared post outside of concrete usage case of editing mutation it would not exactly do that, it is now basically `updateSharePostFromAuthor`. 

So since we have checks on mutation level I would remove `authorId` for that case and make it more reusable. If it makes sense?

Reply:
that's a fair point 👍 updated